dev/src/lobster/flat_gltf.h

```diff
@@ -0,0 +1,789 @@
+    _VERBOUT_ << "Accessors:    " << accessors.size() << std::endl << std::endl;
+
+    Scene scene = newScene;
+
+    if (info->materials()) {
+      for (size_t i = 0; i < info->materials()->size(); i++) {
+        auto m = info->materials()->Get(i);
+        SceneMaterial mat;
+
+        if (m->pbrMetallicRoughness()) {
+          auto pbr = m->pbrMetallicRoughness();
+          if (pbr->baseColorTexture()) {
+            int idx = info->textures()
+                          ->Get(pbr->baseColorTexture()->index())
+                          ->source();
+            std::string imgName = info->images()->Get(idx)->uri()->str();
+            _VERBOUT_ << "Loading \"" << imgName << "\" for baseColorTexture"
+                      << std::endl
+                      << std::endl;
+            mat.baseColorImage = LoadImage(gltfPath, imgName);
+          }
+          mat.metallicFactor = pbr->metallicFactor();
+          mat.roughnessFactor = pbr->roughnessFactor();
+        }
+
+        scene.materials.push_back(mat);
+      }
+    }
+
+    if (info->meshes()) {
+      for (size_t i = 0; i < info->meshes()->size(); i++) {
+        auto m = info->meshes()->Get(i);
+        SceneMesh primarr;
+        for (size_t j = 0; j < m->primitives()->size(); j++) {
+          auto p = m->primitives()->Get(j);
+          int mode = p->mode() > 0 ? p->mode() : 4;
+          _VERBOUT_ << "Primitive is " << (mode == 4 ? "" : "not ")
+                    << "made out of triangles." << std::endl;
+          std::vector<fgl::Vertex> verts;
+          std::vector<fgl::Index> inds;
+
+          // TODO: maybe verify this
+          // if( p->attributes()->Verify( (((a verifier))) ) ){
+          // TODO: the 0-indexed accessors array probably results in bogus
+          // values for undefined attributes FlatBuffers sets undefined values
+          // to 0 if undefined attributes aren't used, it's fine
+          int atIndices[4] = {
+              p->attributes()->POSITION(), p->attributes()->NORMAL(),
+              p->attributes()->TANGENT(), p->attributes()->TEXCOORD_0()};
+
+          for (int atType = 0; atType < 4; atType++) {
+            Accessor a = accessors[atIndices[atType]];
+            int count = a.count;
+            BufferView bv = bufferviews[a.bufferView];
+            int bStride = bv.stride;
+            // According to the glTF 2.0 spec:
+            //   "When `byteStride` of referenced `bufferView` is not defined,
+            //   it means that... effective stride equals the size of the
+            //   element."
+            if (bStride == 0 && a.type == "VEC3") {
+              // 3 components * 4 bytes
+              bStride = 12;
+            } else if (bStride == 0 && a.type == "VEC2") {
+              // 2 components * 4 bytes
+              bStride = 8;
+            }
+            // we only use floats for vertex attributes
+            if (a.componentType != 5126)
+              count = 0;
+            if (count > (int)verts.size())
```

Comment:
make `count` a `size_t` or whatever requires the least amount of casts.

Reply:
glTF does not use unsigned ints for counts, and that is the only cast in a comparison with count, so it's fine as an int. Anyhow, aren't the C++ people trying to phase out size_t?